modules_core/com.smf.securewebservices/src/com/smf/securewebservices/utils/SecureWebServicesUtils.java

```diff
@@ -270,84 +364,25 @@ public static String generateToken(User user, Role role, Organization org, Wareh
-					if (userRole.getRole().getId().equals(role.getId())) {
-						selectedRole = role;
-						break;
-					}
-				}
-
-			// if user has
-			if (selectedRole == null) {
-				if (defaultWsRole != null) {
-					selectedRole = defaultWsRole;
-				} else if (defaultRole != null) {
-					selectedRole = defaultRole;
-				} else if (userRoleList.size() > 0) {
-					selectedRole = userRoleList.get(0).getRole();
-				} else {
-					log.error("SWS - The selected user (\"" + user.getId() + "\") has no roles");
-					throw new Exception(Utility.messageBD(new DalConnectionProvider(), "SMFSWS_UserHasNoRole",
-							OBContext.getOBContext().getLanguage().getLanguage()));
-				}
-			}
-			List<RoleOrganization> roleOrgList = selectedRole.getADRoleOrganizationList();
-			// if organization is valid, select
-			if (org != null)
-				for (RoleOrganization roleOrg : roleOrgList) {
-					if (roleOrg.getOrganization().getId().equals(org.getId())) {
-						selectedOrg = org;
-						break;
-					}
-				}
-			// if not valid select default org for the selected role
-			if (selectedOrg == null) {
-				if (defaultRole != null && defaultRole.getId().equals(selectedRole.getId()) && defaultOrg != null) {
-					selectedOrg = defaultOrg;
-				} else if (roleOrgList.size() > 0) {
-					selectedOrg = roleOrgList.get(0).getOrganization();
-				} else {
-					log.error("SWS - The selected role (\"" + selectedRole.getId() + "\") has no organization");
-					throw new Exception(Utility.messageBD(new DalConnectionProvider(), "SMFSWS_RoleHasNoOrg",
-							OBContext.getOBContext().getLanguage().getLanguage()));
-				}
-			}
+			selectedRole = getRole(role, userRoleList, defaultWsRole, defaultRole);
+			selectedOrg = getOrganization(org, selectedRole, selectedOrg, defaultRole, defaultOrg);
+			selectedWarehouse = getWarehouse(warehouse, selectedOrg, selectedWarehouse, defaultWarehouse);
 
-			List<Warehouse> warehouseList = SecureWebServicesUtils.getOrganizationWarehouses(selectedOrg);
-			// if warehouse is valid, select
-			if (warehouse != null)
-				for (Warehouse wh : warehouseList) {
-					if (wh.getId().equals(warehouse.getId())) {
-						selectedWarehouse = warehouse;
-						break;
-					}
-				}
-			// if not valid select default warehouse for the selected org
-			if (selectedWarehouse == null) {
-				if (defaultWarehouse != null) {
-					selectedWarehouse = defaultWarehouse;
-				} else if (warehouseList.size() > 0) {
-					selectedWarehouse = warehouseList.get(0);
-				} else {
-					log.error("SWS - The selected organization (\"" + selectedOrg.getId() + "\") has no warehouses");
-					throw new Exception(Utility.messageBD(new DalConnectionProvider(), "SMFSWS_OrgHasNoRole",
-							OBContext.getOBContext().getLanguage().getLanguage()));
-				}
+			Algorithm algorithm;
+			if (StringUtils.equals("RS256", config.getAlgorithm())) {
+				final PrivateKey privateKey = getPrivateKey(config);
+				algorithm = Algorithm.RSA256((RSAPrivateKey) privateKey);
+			} else {
```

Comment:
Only these two algorithms can be used? Other devs can't define another kinds of them?

Reply:
Not yet... For now the only two allowed are HS256 and RS256... Is planned to make it configurable so another devs can add more algorithms.